src/object/cli_shard.c

```diff
@@ -1125,7 +1227,9 @@ dc_obj_shard_rw(struct dc_obj_shard *shard, enum obj_rpc_opc opc,
 	rw_args.co = shard->do_co;
 	rw_args.shard_args = args;
 	/* remember the sgl to copyout the data inline for fetch */
-	rw_args.rwaa_sgls = (opc == DAOS_OBJ_RPC_FETCH) ? sgls : NULL;
+	rw_args.rwaa_sgls = sgls;
+	rw_args.send_time = daos_get_ntime();
```

Comment:
calling this on the regular path (no metrics enabled) sounds like a bad idea IMO and should be avoided.

Reply:
ah, sure

Reply:
are you going to change this or fix in following patch?

Reply:
This has been fixed by the following commit in PR , not sure why it is not shown here. hmm
[
](https://github.com/daos-stack/daos/pull/13517/commits/8182b70a22a0fae570c438ae3cb29cd7c3940d56)